engine/src/error.rs

```diff
@@ -0,0 +1,21 @@
+use std::fmt;
+
+/// An enum containing all kinds of game engine errors.
+#[derive(Debug, PartialEq)]
+pub enum EngineError {
+    InvalidEntityId(i32),
+    OutOfMapCoordinate(f32, f32),
+    PositionOccupied(f32, f32),
+}
```

Comment:
It seems like we use this type in the context of processing actions. We take `ActionData`, process it and return a `EngineResult` rather than an `ActionResult`. I am probably missing the other uses, but what other kinds of `EngineError`s unrelated to this use case do you anticipate?

Reply:
Also not seeing none. Good idea no moving this to the action context. Even if we do need this top level error group, we can bundle it together. This way we can group errors by context.